common/include/pcl/common/tuple.h

```diff
@@ -0,0 +1,151 @@
+void
+for_each_until_true(Tuple&& t, Function f, std::integral_constant<size_t, I>)
+{
+  bool exit = f(std::get<I>(std::forward<Tuple>(t)));
+
+  if (!exit)
+    for_each_until_true(
+        std::forward<Tuple>(t), f, std::integral_constant<size_t, I + 1>());
+}
+} // namespace detail
+
+/**
+ * \brief Iterates over all elements of tuples until the function called returns true
+ *
+ * \tparam Tuple The tuple to iterate through
+ * \tparam Function A callable that is invoked for every tuple element and returns a
+ * boolean indicating whether to continue iteration or not
+ *
+ * \remark Implementation taken from
+ * https://stackoverflow.com/questions/26902633/how-to-iterate-over-a-stdtuple-in-c-11
+ */
+template <typename Tuple, typename Function>
+void
+for_each_until_true(Tuple&& t, Function f)
+{
+  detail::for_each_until_true(std::forward<Tuple>(t), f, std::integral_constant<size_t, 0>());
+}
+
+namespace detail {
+
+template <typename T, typename Tuple>
+struct tuple_contains_type_impl;
+
+template <typename T, typename... Us>
+struct tuple_contains_type_impl<T, std::tuple<Us...>>
+: pcl::disjunction<std::is_same<T, Us>...> {};
+
+} // namespace detail
+
+/**
+ * \brief If the \Tuple contains the type \p T then provides the member constant value
+ * equal true. For any other type, value is false.
+ * *
+ * \tparam T a type to check
+ * \tparam Tuple a tuple in which to check for the type
+ *
+ */
+template <typename T, typename Tuple>
+using tuple_contains_type = detail::tuple_contains_type_impl<T, Tuple>;
+
+template <typename T, typename Tuple>
+constexpr bool tuple_contains_type_v = detail::tuple_contains_type_impl<T, Tuple>::value;
+
+static_assert(tuple_contains_type_v<int, std::tuple<int, float, double, unsigned>>,
+              "Failed to check type in tuple");
+
+namespace detail {
+
+template <template <typename...> class Predicate, typename... TupleElements>
+struct filter_tuple_values_impl {
+  using type = decltype(std::tuple_cat(
+      typename std::conditional<Predicate<TupleElements>::value, std::tuple<TupleElements>, std::tuple<>>::
+          type()...));
+
+  /**
+   * \brief Checks whether a tuple contains a specified type
+   *
+   * \tparam TupleElements the elements of the tuple you want to filter
+   * \param std::tuple<TupleElements...> a tuple of the elements you want to filter
+   * \return a tuple containing the filtered elements
+   *
+   */
+  auto
+  operator()(const std::tuple<TupleElements...>& in)
+  {
+    return (*this)(in, type{});
+  }
+
+private:
+  // Utility function to fetch the types we're interest in outputting
+  template <typename... To>
+  auto
+  operator()(const std::tuple<TupleElements...>& in, std::tuple<To...>)
+  {
+    return std::make_tuple(std::get<To>(in)...);
+  }
+};
+
+} // namespace detail
+
+/**
+ * \brief Filters elements of \p Tuple based on the provided predicate/condition
+ *
+ * \tparam Predicate A trait which takes a tuple element as parameter and defines a
```

Comment:
Just a minor update needed: the trait takes a tuple element-type, not a tuple element. Update needed in 2 places